src/evaluators/UnaryExpression.js

```diff
@@ -265,7 +265,7 @@ function evaluateOperation(
       return new StringValue(realm, "number");
     } else if (isInstance(proto, SymbolValue)) {
       return new StringValue(realm, "symbol");
-    } else if (isInstance(proto, ObjectValue)) {
+    } else if (isInstance(proto, ObjectValue) && !(val instanceof AbstractValue)) {
```

Comment:
On an `AbstractObjectValue` `IsCallable()` could be true _or_ false. Depending on whether or not the abstract object is callable. Since `IsCallable()` returns a boolean, we throw when there is a “top” `AbstractObjectValue` instead of emitting `typeof x` like the branch below does. I added this condition so that `AbstractObjectValue`s went to the `AbstractValue.createFromUnaryOp(realm, "typeof", val)` branch. I think the test failures are unrelated to the correctness of this change.

I decided to back out of this change, though, since `IsCallable()` already has handling for abstract values which I didn’t realize so that means this change was a regression in what values Prepack could concretize. `IsCallable()` didn’t support the “top” object values I was sending it. Instead I opted to return a simple abstract object type which is clearly not callable and does not regress what `typeof` can concretize.

Reply:
I see. The current code seems wrong then. It should not give up when it can't tell if a abstract object is callable or not. Worth calling out in an issue.